packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,167 @@
+# Banner
+
+## Overview
+
+The banner is a component used to display a persistent notification to a user.
+
+### Background
+
+[Design specification](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/29c405f7-08ea-48b6-973f-546970b9dbab/)
+
+[GitHub issue](https://github.com/ni/nimble/issues/305)
+
+### Non-goals
+
+- Arbitrary HTML content
+- A container component that would enforce design constraints on multiple banners displayed together
+- Limitating length of displayed text and/or truncating text with ellipsis
+
+### Features
+
+- Optionally dismissable
+- Optional link or button (only 1) to perform an action/navigation
+- Distinct designs for banners of type Error, Warning, Info, or Neutral
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+Refer to examples in the "Some examples of screenshots" section of the [GitHub issue](https://github.com/ni/nimble/issues/305).
+
+---
+
+## Design
+
+The banner is a standalone component that spans the width of its containing element, presenting, from left to right, an icon (dependent on Error/Warning/Info type), an optional title, an optional plaintext message, an optional action button or link, and an optional close control. The banner is intended to display short messages (recommended three lines or fewer) that communicate some persistent state or condition.
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*: `nimble-banner`
+- *Props/Attrs*:
+  - `heading` - if set, a heading with the specified text is shown to the left of the message. OPEN QUESTION: should we instead use the existing `title` attribute?
+  - `text` - the message text
+  - `prevent-dismiss` - set to hide the close button (attr name taken from Nimble dialog)
+  - `type` - one of `error`, `warning`, `info`, or `neutral` (default)
+  - `action-text` - the text of the action link/button. If unset, no link/button is displayed.
+  - `action-href` - if set, the action element will be a link with the given `href`. (Otherwise, when `action-text` is set, a button element will be displayed.)
+  - `action-button-appearance` - either `outline` or `ghost` (default). When an action button is displayed, this controls the appearance variant.
+- *Methods*
+- *Events*
+  - `close` - fired when the banner is dismissed
+  - `action` - fired when the action button is activated (by mouse or keyboard). This is the intended way for clients to perform an action.
+- *CSS Classes and CSS Custom Properties that affect the component*
+
```

Comment:
I've added an Alternatives section that lists the the two items you suggested. I'm interested to see if anyone likes one of those options better.

Reply:
Is another option to take an approach like we have with the various anchor representations for some components, where we could provide `nimble-banner-button` and `nimble-banner-link` as separate components? This will at least allow us to provide constrained APIs that make sense for each banner representation, and each could still inherit from a common base class (i.e. `NimbleBanner`). If there were ever a need for another type of banner, we would have the pattern in place to introduce a new component type.

There doesn't seem much of a downside of this approach from an Angular (as an example) integrator, where they would just introduce a different component into the template (via *ngIf on some configuration potentially) instead of setting a different property on the same banner element type (via the same *ngIf logic).

Reply:
Note that my suggestion would only make sense if we went against having the content slotted, which might depend on how the conversation @mollykreis started below is resolved. Note that I currently would rather us end up with the action content slotted.

Reply:
I think I've been sold on making the content slotted. Though I still would rather the user slot a standard `nimble-button` or `nimble-anchor` and have our CSS style them specially for the banner.